### PR TITLE
Fix with meta flag for analytics queries

### PR DESCRIPTION
**Pull request by Westwooo**
Addresses: https://github.com/couchbaselabs/couchbase-shell/issues/556

This change means that the results of an analytics query will not be streamed when the with-meta flag is used, this behaviour will need to be documented. It would be possible to have the meta returned and still stream the queries, but this has two issues: 

1) The output table would be mostly empty as all the rows will have the metadata columns but these would all be empty
2) The purpose of streaming results is to prevent reading them to memory by streaming to a file, but the format of the metadata means that it cannot be saved to file in a streaming manner.

### CODE REVIEW

File: src/cli/analytics.rs [hunk 1]
```diff
@@ -164,11 +167,60 @@ fn run(
...
+                .map_err(|e| ShellError::GenericError {
+                    error: format!("failed to read analytics query result: {}", e),
+                    msg: "".to_string(),
+                    span: None,
+                    help: None,
+                    inner: vec![],
+                })?
+            {
+                if !query_results.is_empty() {
+                    query_results.append(&mut vec![b','])
```
Comment (chvck) [inline src/cli/analytics.rs:183]: Why is this needed?

File: src/cli/analytics.rs [hunk 2]
```diff
@@ -164,11 +167,60 @@ fn run(
...
+                .block_on(async { json_streamer.read_epilog().await })
+                .map_err(|e| ShellError::GenericError {
+                    error: format!("failed to read stream epilog: {}", e),
+                    msg: "".to_string(),
+                    span: None,
+                    help: None,
+                    inner: vec![],
+                })?;
+
+            let (start, _) = from_utf8(&epilog).unwrap().split_at(epilog.len() - 1);
```
Comment (chvck) [inline src/cli/analytics.rs:198]: Is there no way that we can create the nu value without manually creating a JSON value?

File: src/cli/analytics.rs (hunk 1 above)
Comment (Westwooo) [reply src/cli/analytics.rs:183]: So that the results are formatted as a list

**Review by chvck — APPROVED** [review]
